{"text": "2023-12-15 06:25:18 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:00.862844", "seconds": 0.862844}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3696, "name": "MainProcess"}, "thread": {"id": 140477124545408, "name": "MainThread"}, "time": {"repr": "2023-12-15 06:25:18.001093+08:00", "timestamp": 1702592718.001093}}}
{"text": "2023-12-15 06:25:23 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Destiny 2: Legacy Collection', 'url': 'https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023'}\n", "record": {"elapsed": {"repr": "0:00:06.448420", "seconds": 6.44842}, "exception": null, "extra": {"title": "Destiny 2: Legacy Collection", "url": "https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023"}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3696, "name": "MainProcess"}, "thread": {"id": 140477124545408, "name": "MainThread"}, "time": {"repr": "2023-12-15 06:25:23.586669+08:00", "timestamp": 1702592723.586669}}}
{"text": "2023-12-15 06:25:28 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023'}\n", "record": {"elapsed": {"repr": "0:00:11.148500", "seconds": 11.1485}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3696, "name": "MainProcess"}, "thread": {"id": 140477124545408, "name": "MainThread"}, "time": {"repr": "2023-12-15 06:25:28.286749+08:00", "timestamp": 1702592728.286749}}}
{"text": "2023-12-15 06:26:04 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023'}\n", "record": {"elapsed": {"repr": "0:00:47.223384", "seconds": 47.223384}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3696, "name": "MainProcess"}, "thread": {"id": 140477124545408, "name": "MainThread"}, "time": {"repr": "2023-12-15 06:26:04.361633+08:00", "timestamp": 1702592764.361633}}}
{"text": "2023-12-15 06:26:40 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023'}\n", "record": {"elapsed": {"repr": "0:01:23.666933", "seconds": 83.666933}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/destiny-2--legacy-collection-2023"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3696, "name": "MainProcess"}, "thread": {"id": 140477124545408, "name": "MainThread"}, "time": {"repr": "2023-12-15 06:26:40.805182+08:00", "timestamp": 1702592800.805182}}}
{"text": "2023-12-15 06:27:17 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3696), thread 'MainThread' (140477124545408): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 88, in async_wrapped\n    return await fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 47, in __call__\n    do = self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 325, in iter\n    raise retry_exc.reraise()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 158, in reraise\n    raise self.last_attempt.result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 50, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py\", line 320, in claim_weekly_games\n    await page.click(\"//button//span[text()='Check Out']\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 10117, in click\n    await self._impl_obj.click(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_page.py\", line 694, in click\n    return await self._main_frame.click(**locals_to_params(locals()))\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py\", line 484, in click\n    await self._channel.send(\"click\", locals_to_params(locals()))\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py\", line 62, in send\n    return await self._connection.wrap_api_call(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py\", line 492, in wrap_api_call\n    return await cb()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py\", line 100, in inner_send\n    result = next(iter(done)).result()\nplaywright._impl._errors.TimeoutError: Timeout 30000ms exceeded.\n", "record": {"elapsed": {"repr": "0:02:00.137042", "seconds": 120.137042}, "exception": {"type": "TimeoutError", "value": "Timeout 30000ms exceeded.", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3696), thread 'MainThread' (140477124545408):", "module": "claim", "name": "__main__", "process": {"id": 3696, "name": "MainProcess"}, "thread": {"id": 140477124545408, "name": "MainThread"}, "time": {"repr": "2023-12-15 06:27:17.275291+08:00", "timestamp": 1702592837.275291}}}
